Add virt capacity benchmark test

ygalblum opened the pull request:
## Type of change

- New feature

## Description

Run Virtualization with Storage Capacity Benchmark test

vishnuchalla commented on common.go:
```diff
@@ -70,3 +90,175 @@ func GatherMetadata(wh *workloads.WorkloadHelper, alerting bool) error {
 }
+
+// GenerateSSHKeyPair generates an SSH key pair and saves them to the specified files
+func generateSSHKeyPair(sshKeyPairPath string) (string, string, error) {
```
Can this go into `go-commons` please? Looks like nice function to be reused across our tooling.

vishnuchalla commented on common.go:
```diff
@@ -70,3 +90,175 @@ func GatherMetadata(wh *workloads.WorkloadHelper, alerting bool) error {
+	return getVolumeSnapshotClassNameForProvisioner(provisioner)
+}
+
+func virtctlExists() bool {
```
Some of the functions in this file seem to be commonly used across all the virtualization portifolio. Can we move all of them to go-commons please?

ygalblum commented:
I've updated the code. However, it now depends on changes made in https://github.com/cloud-bulldozer/go-commons/. Once the all changes are merged and the dependency is updated here, this code change will compile successfully

jtaleric commented on cmd/config/virt-capacity-benchmark/virt-capacity-benchmark.yml:
```diff
@@ -0,0 +1,207 @@
+- name: create-vms
+  jobType: create
+  jobIterations: 1
+  qps: 20
```
curious why in the first job you have the CNV default of 5/10 for QPS/Burst -- but here and further down you default to 20/20?

ygalblum replied:
TBH I didn't put too much thought into it. Especially since most of the jobs are pretty small. So, not sure it even matters. Which values would you recommend here?

ygalblum commented:
@vishnuchalla I've added nobs that allow skipping some checks and setting some values so that the test can pass with the CI cluster. Hopefully the test will pass in the CI as well (along with the checks I've added)

ygalblum commented:
The CI is failing because the `BeforeCleanup` script is implemented for `bash` while the default shell in Ubuntu (which is the runner) is `dash`. I'm looking into how to address this. Probably need to address it in `kube-burner`

ygalblum commented:
@vishnuchalla PTAL. I solved the CI issue by changing the script so that it will work also on `dash`.

vishnuchalla reviewed (approved):
lgtm
